drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c

```diff
@@ -268,7 +298,30 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
 
         netdev_rx_info->lqi = rx_info.lqi;
     }
-
+#if !IS_ACTIVE(CONFIG_IEEE802154_AUTO_ACK_DISABLE)
```

Comment:
I wouldn't put this behind a config but behind a pseudo-module that the driver can select if it needs software-ACKs.